### DISCUSSION TITLE
Consulta flujo parte 1

### DISCUSSION THREAD

**Discussion by Mikson16** [IIC2133-PUC/2024-2#174] [Tarea 3] [answered]
Hola, no entiendo muy bien el flujo de la parte 1, los primero 4 inputs que se reciben son, la cantidad de pasos, el tiempo de espera para que aumente en 1 la profundidad del sueño, el umbral mínimo y la profundidad de sueño inicial, luego, se reciben los x pasos indicados al inicio, tengo una pregunta en esta parte, ¿Siempre se espera las k unidades de tiempo entre paso y paso? ¿El pingüino siempre sigue los pasos a medida que van llegando? Me refiero si son 5 pasos los necesarios por ejemplo el ruido asociado a cada paso son: {1, 4, 6, 3, 2}, independiente de si el puffle se despierta o no, ¿siempre sigo los pasos en orden de llegada?, es decir primero el paso con ruido 1, luego con 4, y así sucesivamente. Digo esto porque, ya que son pasos que da el pingüino para cruzar la habitación no me hace sentido que si habiendo dado el primer paso p = p-1, se espera y luego si se da el siguiente paso con ruido 4, el puffle se despierta y si en vez de ese se da el de ruido 6, también se despertará, entonces, ¿Debo retornar inmediatamente -1 y terminar la ejecución? Ya que el puffle se despertara de todas formas

**Comment by JavierLarre** [comment, marked as answer]
Hola! no tienes que esperar por cada paso, puedes dar dos o más pasos seguidos mientras no bajes de la profundidad mínima. Además, siempre tienes que dar los pasos en orden. Por ejemplo: en la serie (1, 4, 6, 3, 2) el primer paso siempre es de peso 1, luego el segundo paso es de peso 4, y así sucesivamente. Tienes que manejar qué hacer (o mejor dicho, como esperar) en los casos donde el paso que vas a dar va a bajar la profundidad menos al mínimo. 

Espero haberte ayudado! puedes enviar otra consulta si aún te sientes confundido, igualmente te recomiendo revisar las otras issues por si aún no tienes claro que hacer.

**Reply by Mikson16** [reply]
¿Entonces debo decidir cuando esperar y cuando no entre pasos? Es decir, entre el primer y segundo paso espero, pero del segundo al tercero puedo decidir no esperar.

**Reply by JavierLarre** [reply]
Así mismo lo puedes hacer! No olvides que existe una profundidad máxima que debes respetar.

**Reply by Mikson16** [reply]
¿Si el factor de recuperación al esperar es k, yo puedo esperar más de una vez entre pasos? Es decir, entre el paso 1 y 2, espero por 2k, recuperando 2k unidades de sueño, por ejemplo.

**Reply by JavierLarre** [reply]
No exactamente, puedes decidir esperar más de una vez si lo consideras necesario. Tienes que esperar k unidades de tiempo para aumentar la profundidad de sueño en 1, así que puedes esperar n*k tiempo para aumentar el sueño en n antes de dar un paso, mientras se respete la profundidad máxima de sueño.

**Reply by Mikson16** [reply]
Yaya, gracias, ahora entendi mejor el problema
